apps/mobile-wallet/src/App.tsx

```diff
@@ -53,12 +53,13 @@ import {
 import { store } from '~/store/store'
 import { selectTransactionUnknownTokenIds } from '~/store/transactions/transactionSelectors'
 import { appLaunchedWithLastUsedWallet } from '~/store/wallet/walletActions'
+import { metadataRestored } from '~/store/wallet/walletSlice'
 import { themes } from '~/style/themes'
 
 SplashScreen.preventAutoHideAsync()
 
 const App = () => {
-  const showAppContent = useShowAppContentAfterValidatingStoredWalletData()
+  const { showAppContent, wasMetadataRestored } = useShowAppContentAfterValidatingStoredWalletData()
```

Comment:
It's actually the mnemonic - or the "wallet" - that is restored, no? Metadata are gone, can't be restored.

Reply:
haha interesting, it depends how you see it! I see it more on the perspective that metadata was not there, so we restored them/recreated them. The mnemonic was always there so no need to restore anything. Since it's strictly a programming variable and not visible to the user, I'd prefer to leave the word "metadata" in it, to help out future selves understand what's going on here with a glace. WDYT?